volunteer.php

```diff
@@ -650,4 +664,62 @@ function volunteer_civicrm_fieldOptions($entity, $field, &$options, $params) {
   if ($entity == "UFJoin" && $field == "entity_table" && $params['context'] == "validate") {
     $options[CRM_Volunteer_DAO_Project::getTableName()] = CRM_Volunteer_DAO_Project::getTableName();
   }
+}
+
+/**
+ * This function fetches the defaults from civicrm settings
+ * And puts them into the appropriate data format to return
+ * to the angular front-end
+ *
+ * @return array
+ * @throws CiviCRM_API3_Exception
+ */
+function volunteer_composeDefaultArray() {
+  $defaults = array();
+  $settings = CRM_Core_BAO_Setting::getItem("volunteer_defaults");
```

Comment:
The behavior you may be expecting was [deprecated in 4.7](https://github.com/civicrm/civicrm-core/commit/3a84c0ab9022feeed11adf8f9023fd5f2d4459ba#diff-eaa4bbbf67c4d03afd071da3748383abR241).

Totten provided some background on the API list:

> [A]s soon as we reworked the setting layer in 4.7, the runtime for tests dropped by ~80% (3hr vs 40min). As part of the new regime, settings are loaded by exactly two queries during bootstrap. Pseudocode:
>
>  SELECT name, value FROM civicrm_setting WHERE domain_id = %1;
>  SELECT value FROM civicrm_cache WHERE path = ‘setting-defaults’;
>
> It doesn’t matter how many times you call the setting functions — it’s still just the two original queries. In theory, the balance could change in the future (eg if the settings size grow by 100x), but the current numbers don’t support those theories. (As a notable reference point… D6+D7 have similar performance profile for variable_get(). It only changed in D8 after they needed to account for massive growth in contrib and after they expanded the scope of ‘config’ to include things which had previously been distinct tables.)
>
> So… the theoretical concern does exist, and it’s reasonable to have some kind of get-by-prefix or get-by-group function in anticipation. For now, it can just be a simple filter around Civi::settings()->all();  but in the future it could be something more complicated.

Alternately, you could fetch them one by one:

```php
$foo = CRM_Core_BAO_Setting::getItem('myGroup', 'foo');
```